### PR TITLE
Add deepToRaw method for recursively converting proxy objects to raw

### LINKED ISSUE #11541: Vue 3: Web Worker communication fails due to Proxy objects in messages
After migrating to Vue3, some messages sent to the web worker would fail with the following error:

```
Uncaught (in promise) DOMException: Proxy object could not be cloned.
    createWorker subscribe.js:182
```

We temporarily resolved this problem by utilizing `cloneDeep()`[^1], but this isn't a sufficient approach for the long term. There are a few potential issues with this approach:

[^1]: https://github.com/rancher/dashboard/blob/vue3-depth-first/shell/plugins/steve/subscribe.js#L184

1. `cloneDeep()` might not be cloning Proxy objects, just copying references
2. Modifying references to proxy object might lead to unexpected behavior
3. Data stored in the web worker might not be fully detached from Vue's reactivity system

## Proposed Solution

There are several approaches that can address this issue, one that come to mind:

- Recursively check properties in a message and convert any proxies to raw objects using `toRaw()` [^2]

[^2]: https://vuejs.org/api/reactivity-advanced.html#toraw

### LINKED ISSUE #12406: `isRancher` value gets set to `false` randomly
**Setup**
- Rancher version: 2.10
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

In initial loading of the app `isRancher` gets set to `false` randomly. This causes to not display projects in the app, for example in the namespace dropdown. `Namespaces/Projects` on the side bar also changes to `Namespaces`. Some e2e tests get failed due to this for example `namespace-picker.spec.ts`.

The problem occurs [here](https://github.com/rancher/dashboard/blob/master/shell/store/index.js#L795) where we get `rejected` for `res.rancherSchemas.status`.

**To Reproduce**

There's no reliable way to reproduce this, but refreshing the explorer page can be a good way to repro, or running e2e tests `namespace-picker.spec.ts`.

**Result**

**Expected Result**

**Screenshots**

`console.log(res, res.rancherSchemas.status, res.rancherSchemas.reason, getters['management/schemaFor'](MANAGEMENT.PROJECT))`:
Screenshot 2024-10-25 at 9 49 08 AM

**Additional context**

Some discussion around this issue has happened on the UI's Slack channel.

### PR DESCRIPTION
**Pull request by jordojordo**
### Summary
Fixes #12406 
Fixes #11541 

In Vue 3, the reactivity system uses proxies to make objects reactive. These proxies cannot be serialized or cloned using the structured clone algorithm, which is why we're encountering the `DataCloneError` when attempting to send data that includes these proxies via `postMessage`.

Unfortunately, it wasn't as simple as just converting the `data` object to a raw object and sending that to `postMessage` as there are nested reactive objects that wouldn't be converted in this way. So I implemented a method that would recursively convert all of the proxy objects into raw using Vue's provided `toRaw` method.

### Occurred changes and/or fixed issues

- Added `deepToRaw` method to the `object` utils
- Before passing the data to loadSchemas, we convert all of the proxied objects to raw
- Replaced usage of lodash's `cloneDeep` when sending a message through `postMessage` to a store's worker

### Areas or cases that should be tested

Testing this is a little challenging, the only way to reproduce the issue locally was by constantly refreshing the page until the `Projects` label was missing from `Projects/Namespaces` menu item. When only the `Namespaces` menu item existed, that was caused due to the `rancherSchemas` status being `rejected` when initializing the `isRancher` variable within the `loadManagement` store action.

### Areas which could experience regressions

I'm not sure if this will have a regression effect anywhere as the original comment above the `worker.postMessage` hook implied that the data was to be considered `raw` already. This new method only ensures that's the case by removing proxied objects.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Comment by rak-phillip** [comment]
@jordojordo we should be able to resolve an open TODO[^1] and close out #11541 with this change. Can we look at making the change replace the previous `_.cloneDeep` usage?

[^1]: https://github.com/rancher/dashboard/pull/11686/commits/03e33c7c29392d80f60c10112a8f0ef3a9bc0e63

File: shell/utils/__tests__/object.test.ts
Comment (rak-phillip) [inline shell/utils/__tests__/object.test.ts:1]: I think that we want to add a test case for objects with nested reactive properties, something along the lines of:

```
{
  num:   1,
  str:   'test',
  bool:  reactive(true),
  nil:   null,
  undef: undefined,
  arr:   [1, 2, { a: 3 }],
  obj:   { nested: reactive({ a: 1 }) },
  func:  null,
  sym:   null,
}
```

**Comment by jordojordo** [comment]
@rak-phillip Ah sorry, I didn't notice that TODO. I'll replace it and test it out.

**Review by rak-phillip — APPROVED** [review]
LGTM - I'm unable to reproduce the issues described in #11541 & #12406. Based on the transient behavior described in #12406, I think that it would make sense to one more approval before merging.

**Review by codyrancher — APPROVED** [review]

**Comment by momesgin** [comment]
I couldn't reproduce the issue anymore

**Pull request merged**